rasa/cli/export.py

```diff
@@ -0,0 +1,217 @@
+
+    Return:
+        List of conversation IDs requested as a command-line argument.
+        `None` if that argument was left unspecified.
+
+    """
+    if not conversation_ids_arg:
+        return None
+
+    return conversation_ids_arg.split(",")
+
+
+def _validate_timestamp_options(args: argparse.Namespace) -> None:
+    """Inspect CLI timestamp parameters.
+
+    Prints an error and exits if a maximum timestamp is provided that is smaller
+    than the provided minimum timestamp.
+
+    Args:
+        args: Command-line arguments to process.
+
+    """
+    min_timestamp = args.minimum_timestamp
+    max_timestamp = args.maximum_timestamp
+
+    if (
+        min_timestamp is not None
+        and max_timestamp is not None
+        and max_timestamp < min_timestamp
+    ):
+        cli_utils.print_error_and_exit(
+            f"Maximum timestamp '{max_timestamp}' is smaller than minimum "
+            f"timestamp '{min_timestamp}'. Exiting."
+        )
+
+
+def _prepare_event_broker(event_broker: EventBroker) -> None:
+    """Sets `should_keep_unpublished_messages` flag to `False` if
+    `self.event_broker` is a `PikaEventBroker`.
+
+    If publishing of events fails, the `PikaEventBroker` instance should not keep a
+    list of unpublished messages, so we can retry publishing them. This is because
+    the instance is launched as part of this short-lived export script, meaning the
+    object is destroyed before it might be published.
+
+    In addition, wait until the event broker reports a `ready` state.
+
+    """
+    if isinstance(event_broker, PikaEventBroker):
+        event_broker.should_keep_unpublished_messages = False
+        event_broker.raise_on_failure = True
+
+    if not event_broker.is_ready():
+        cli_utils.print_error_and_exit(
+            f"Event broker of type '{type(event_broker)}' is not ready. Exiting."
+        )
+
+
+def export_trackers(args: argparse.Namespace) -> None:
+    """Export events for a connected tracker store using an event broker.
+
+    Args:
+        args: Command-line arguments to process.
+
+    """
+    _validate_timestamp_options(args)
+
+    endpoints = rasa_core_utils.read_endpoints_from_path(args.endpoints)
+    tracker_store = _get_tracker_store(endpoints)
+    event_broker = _get_event_broker(endpoints)
+    _prepare_event_broker(event_broker)
+    requested_conversation_ids = _get_requested_conversation_ids(args.conversation_ids)
+
+    migrator = Migrator(
+        tracker_store,
+        event_broker,
+        args.endpoints,
+        requested_conversation_ids,
+        args.minimum_timestamp,
+        args.maximum_timestamp,
+    )
+
+    try:
+        published_events = migrator.publish_events()
+        cli_utils.print_success(
+            f"Done! Successfully published {published_events} events 🎉"
+        )
+
+    except PublishingError as e:
+        command = _get_continuation_command(migrator, float(e))
```

Comment:
what is this `float` cast? Can't we set a property on the exception?

Reply:
the exception message is still there (timestamp as a string), the float is for convenience